posixlib/src/main/resources/scala-native/time.c

```diff
@@ -55,6 +45,22 @@ static void tm_init(struct tm *tm, struct scalanative_tm *scala_tm) {
     tm->tm_wday = scala_tm->tm_wday;
     tm->tm_yday = scala_tm->tm_yday;
     tm->tm_isdst = scala_tm->tm_isdst;
+
+    if (sizeof(struct tm) > sizeof(struct scalanative_tm)) {
+      // The operating system struct tm can be larger than
+      // the scalanative tm.  On 64 bit GNU or _BSD_SOURCE Linux this
+      // usually is true and beyond easy control.
+      //
+      // Clear any fields not known to scalanative, such as tm_zone,
+      // so they are zero/NULL, not J-Random garbage.
+      // strftime() in Scala Native release mode is particularly sensitive
+      // to garbage beyond the end of the scalanative tm.
+      // Assume all excess size is at bottom of C tm, not internal padding.
+
+      char *start = (char *) tm + sizeof(struct scalanative_tm);
+      size_t count = sizeof(struct tm) - sizeof(struct scalanative_tm);
+      memset(start, 0, count);
```

Comment:
Let image that some wired system adds some new field *before* or somewhere in the middle of this structure.

Reply:
Yes, nasty code and you do dredge up nightmares.  I ran out of time tonight but perhaps we can
come up with something better or at least something that is likely to make the test fail in
an obvious manner. Only to have some future devo, rightly, rant at me for writing such
fragile code.

In normal code I believe that C has a way of getting the offset of a named field.
Here I have failed so far to find a way to get the named field on systems 
where it is not defined (i.e. short of 36 byte).  The C offsetof (?) was developed
for exactly the situations you describe.  I will think on what might be possible.

Reply:
A quick thought.   "All things to all systems" is a good goal but hard with finite resources.
Perhaps I should make the assumption of a *> SN 36 byte is 56 byte, no padding" an 
explicit assertion at the beginning of the Test.  That is still susceptible to re-arrangement
of the fields on weird systems, but at least hard fails when the assumption is false.

I need to give this more thought.

Agreed, magic numbers like 36 & 56 are a code smell, like being downwind of a pig farm.

Reply:
@LeeTibbert I do have a simpel idea: current code defines `struct tm` alway on stack. Why can't you use `struct tm = {0}` to ask compiler to make `memset`?